ludwig/schema/features/preprocessing/bag.py

```diff
@@ -55,9 +55,20 @@ class BagPreprocessingConfig(BasePreprocessingConfig):
     )
 
     most_common: int = schema_utils.PositiveInteger(
-        default=10000,
+        default=None,
         allow_none=True,
         description="The maximum number of most common tokens to be considered. If the data contains more than this "
         "amount, the most infrequent tokens will be treated as unknown.",
         parameter_metadata=FEATURE_METADATA[BAG][PREPROCESSING]["most_common"],
     )
+
+    most_common_percentile: int = schema_utils.FloatRange(
+        default=1,
+        min=0.0,
+        max=1.0,
+        min_inclusive=False,
+        allow_none=False,
```

Comment:
How does this interact with `most_common` if we can't set to null?

Reply:
Add `most_common_percentile` parameter in addition to `most_common`. Specifying `most_common_percentile: 0.9` will keep the top 90% most frequent tokens in the vocab. `most_common` is set to `None` by default. If both `most_common` and `most_common_percentile` are set, `most_common` overrides.